dashboard/lib/services/lti.rb

```diff
@@ -160,31 +171,53 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_members)
       had_changes = false
       client_id = lti_integration.client_id
       issuer = lti_integration.issuer
-      current_students = nrps_members.map do |nrps_member|
-        student = Queries::Lti.get_user_from_nrps(
+      section = lti_section.section
+      current_students = Set.new
+      current_teachers = Set.new
+
+      nrps_members.each do |nrps_member|
+        account_type = Policies::Lti.get_account_type(nrps_member[:roles])
+
+        user = Queries::Lti.get_user_from_nrps(
           client_id: client_id,
           issuer: issuer,
           nrps_member: nrps_member
         )
-        student ||= initialize_lti_student_from_nrps(
+        user ||= initialize_lti_user_from_nrps(
           client_id: client_id,
           issuer: issuer,
           nrps_member: nrps_member
         )
-        had_changes ||= (student.new_record? || student.changed?)
-        student.save!
-        add_student_result = lti_section.section.add_student(student)
-        had_changes ||= add_student_result == Section::ADD_STUDENT_SUCCESS
-        student
+        had_changes ||= (user.new_record? || user.changed?)
+        user.save!
+        if account_type == ::User::TYPE_STUDENT
+          add_student_result = section.add_student(user)
+          had_changes ||= add_student_result == Section::ADD_STUDENT_SUCCESS
+          current_students.add(user.id)
+        else
+          add_instructor_result = section.add_instructor(user)
+          had_changes ||= add_instructor_result
+          current_teachers.add(user.id)
+        end
       end
 
       # Prune students who have been removed from the section in the LMS
       lti_section.followers.each do |follower|
-        unless current_students.find_index {|s| s.id == follower.student_user_id}
+        unless current_students.include?(follower.student_user_id)
           follower.destroy
           had_changes = true
         end
       end
+
+      # Prune teachers who have been removed from the section in the LMS
+      section.section_instructors.each do |section_instructor|
+        instructor = section_instructor.instructor
+        unless current_teachers.include?(instructor.id)
+          section.remove_instructor(instructor)
```

Comment:
If `remove_instructor` no longer touches the section owner this is going to silently leave the teacher who was removed from the LMS as a teacher on our platform a rather large percentage of the time. Reassigning the section ownership to one of the remaining instructors and then removing this instructor would probably be wise.

Reply:
Sure, sounds good. I also think the idea of just auto-promoting the one triggering the sync instead of making a special dialog to ask the teacher would be fine as well.